### ISSUE TITLE
Train on non-face images

### ISSUE THREAD

**Issue by kail85** [eladrich/pixel2style2pixel#269]
It looks like the train script assumes input images are face images, so it calls the IR-SE50 Model. Could you please share some insights about training on non-face images, for example, medical images. What kind of modification shall I apply on the training script? Thank you.

**Comment by yuval-alaluf** [comment]
You're correct that the encoder is trained with a network trained for facial recognition. One option is to change the backbone to something that is pretrained on ImageNet and is based on say ImageNet. For this, you can refer to this class in my ReStyle encoder repo:
https://github.com/yuval-alaluf/restyle-encoder/blob/3dacd706c79f42fc828afc0d3b05f29dd0406454/models/encoders/fpn_encoders.py#L85

You can simply change the call from the current GradualStyleEncoder to this ResNetGradualStyleEncoder and I believe it should work for you. 

Other than that, we use an ID loss that uses ArcFace. You can either turn that off by setting the loss lambda to 0, or swap that loss with a MoCo-based loss by setting `moco_lambda` to 0.5 for example. Since you're working with medical imagery, I am not sure how much the MoCo loss will help you here, but you can try it out.

**Comment by TUCK-goethe** [comment]
Hello,
First of all, I want to give a big thumbs up for the repository. For my work I need to generate images of old coins. For this I added the ResNetGradualStyleEncoder class to the psp_encoders file as described above. For load_weights Im using the resnet34 model. However, when training, I still recognize faces in the outputs, but I don't know where the faces are coming from. 
My input is the following:
python scripts/train.py 
--dataset_type=coins 
--exp_dir=path_to_experiment 
--workers=2 --batch_size=2 --test_batch_size=2 --test_workers=2 
--val_interval=2500 --save_interval=5000 --encoder_type=ResNetGradualStyleEncoder --start_from_latent_avg --lpips_lambda=0.8 --l2_lambda=1 --moco_lambda=0.5 

Output looks like this:
![0075](https://user-images.githubusercontent.com/67458084/180650954-7421df98-fb8f-4cbb-bb46-700314ff1f9d.jpg)

Do you have any idea what this could be and how i can solve the problem?
Thank you
